openfeature-provider/go/grpc_wasm_flag_logger.go

```diff
@@ -0,0 +1,160 @@
+package confidence
+
+import (
+	"context"
+	"log"
+	"sync"
+	"time"
+
+	resolverv1 "github.com/spotify/confidence-resolver-rust/openfeature-provider/go/confidence/proto/confidence/flags/resolverinternal"
+)
+
+const (
+	// MaxFlagAssignedPerChunk is the max number of flag_assigned entries per chunk
+	// to avoid exceeding gRPC max message size
+	MaxFlagAssignedPerChunk = 1000
+)
+
+// WasmFlagLogger is an interface for writing flag logs
+type WasmFlagLogger interface {
+	Write(ctx context.Context, request *resolverv1.WriteFlagLogsRequest) error
+	Shutdown()
+}
+
+// FlagLogWriter is a function type for writing flag logs
+type FlagLogWriter func(ctx context.Context, request *resolverv1.WriteFlagLogsRequest) error
+
+// GrpcWasmFlagLogger implements WasmFlagLogger using gRPC
+type GrpcWasmFlagLogger struct {
+	stub   resolverv1.InternalFlagLoggerServiceClient
+	writer FlagLogWriter
+	wg     sync.WaitGroup
+}
+
+// NewGrpcWasmFlagLogger creates a new GrpcWasmFlagLogger
+func NewGrpcWasmFlagLogger(stub resolverv1.InternalFlagLoggerServiceClient) *GrpcWasmFlagLogger {
+	logger := &GrpcWasmFlagLogger{
+		stub: stub,
+	}
+
+	// Set up the default writer that sends requests asynchronously
+	logger.writer = func(ctx context.Context, request *resolverv1.WriteFlagLogsRequest) error {
+		logger.wg.Add(1)
+		go func() {
+			defer logger.wg.Done()
+			// Create a context with timeout for the RPC
+			rpcCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
probably just a nit: to my understanding is usually better to use `ctx` rather than creating a new detached background context here. AI suggests me that, if we really want to have a detached context, a slightly better way is `detachedCtx := context.WithoutCancel(ctx)` (but I haven't tested this)